integration_tests/test_link_fetching.py

```diff
@@ -33,7 +33,8 @@ def test_that_link_fetching_invocation_executes_correctly(
     queue_url: str,
 ):
     execution_arn = step_function_client.start_execution(
```

Comment:
Rather than fiddling with the mocked responses I figured we can _just_ request platforms that we know are supported by the mocked API. This will also help us avoid an error when ESA launches Sentinel-2D in >=2028 😄

Probably more importantly this also helps us poke to make sure our StepFunction can handle overriding the "platform" to the "date generator"